src/pybind/mgr/cephadm/inventory.py

```diff
@@ -427,4 +429,74 @@ def add_daemon(self, host, dd):
     def rm_daemon(self, host, name):
         if host in self.daemons:
             if name in self.daemons[host]:
-                del self.daemons[host][name]
\ No newline at end of file
+                del self.daemons[host][name]
+
+class EventStore():
+    def __init__(self, mgr):
+        # type: (CephadmOrchestrator) -> None
+        self.mgr: CephadmOrchestrator = mgr
+        self.events = {} # type: Dict[str, List[OrchestratorEvent]]
+
+    def load(self):
+        # type: () -> None
+        for k, v in self.mgr.get_store_prefix(EVENT_STORE_PREFIX):
+            kind_subject = k[len(EVENT_STORE_PREFIX):]
+            try:
+                v = json.loads(v)
+                events = ServiceSpec.from_json(v.get('evs', []))
+                self.events[kind_subject] = events
+
+                self.mgr.log.debug('SpecStore: loaded events for %s' % (
+                    kind_subject))
+            except Exception as e:
+                self.mgr.log.exception('unable to load events for %s: %s' % (
+                    kind_subject, e))
+                pass
+
+    def add(self, event: OrchestratorEvent) -> None:
+
+        if event.kind_subject() in self.events:
+            self.events[event.kind_subject()].append(event)
+
+            # limit to five events for now.
+            self.events[event.kind_subject()] = self.events[event.kind_subject()][-5:]
+        else:
+            self.events[event.kind_subject()] = [event]
+
+        self.mgr.set_store(
+            EVENT_STORE_PREFIX + event.kind_subject(),
+            json.dumps({
+                'evs': [ev.to_json() for ev in self.events[event.kind_subject()]]
```

Comment:
Can we use `events` here to make it more clear?
```suggestion
                'events': [ev.to_json() for ev in self.events[event.kind_subject()]]
```